### PR TITLE
Opr 107 failed rapportage extend

### PR DESCRIPTION
**Pull request by LiekeSchoenmaker**
Update logica of failed reason database

### CODE REVIEW

**Review by mvanniekerkHartwig — APPROVED** [review]
I understand why you wanted to refactor this to a separate file :)
Both ways are not quite ideal. Approved for now, with some suggestions.

File: patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/panel/PanelFailReason.java [hunk 1]
```diff
@@ -10,16 +10,24 @@
 
 public enum PanelFailReason {
 
-    PANEL_FAILURE("insufficient_dna_panel", QsFormNumber.FOR_345);
+    PANEL_FAILURE("insufficient_dna_panel", QsFormNumber.FOR_345, "Insufficient quality of received biomaterial(s)", "Next Generation Sequencing could not be successfully performed on the received biomaterial(s) because not enough DNA was present after DNA isolation. This is likely due to poor quality of the received biomaterials(s).");
```
[the remark this answers is not in the document]
Comment (LiekeSchoenmaker) [reply patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/panel/PanelFailReason.java:13]: yes, in the future when we will implement a real oncopanel report

File: patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/qcfail/QCFailReason.java [hunk 2]
```diff
@@ -9,11 +9,29 @@
...
+    ISOLATION_FAIL("isolation_fail", QCFailType.LOW_QUALITY_BIOPSY, true, QsFormNumber.FOR_083,
+            "Insufficient quality of received biomaterial(s)", "Whole Genome Sequencing could not be successfully performed on the " +
+            "received biomaterial(s) because not enough DNA was present after DNA isolation. This is likely due to poor quality of the received biomaterials(s)."),
+    TCP_SHALLOW_FAIL("tcp_shallow_fail", QCFailType.LOW_QUALITY_BIOPSY, false, QsFormNumber.FOR_100,
+            "Insufficient quality of received biomaterial(s)","Whole Genome Sequencing could not be successfully performed on the " +
+            "received biomaterial(s) because shallow sequencing analysis showed the tumor cell percentage was too low."),
+    PREPARATION_FAIL("preparation_fail", QCFailType.LOW_QUALITY_BIOPSY, true, QsFormNumber.FOR_100,
+            "Insufficient quality of received biomaterial(s)", "Whole Genome Sequencing could not be successfully performed on " +
+            "the received biomaterial(s) because not enough DNA was present after sample preparation. This is likely due to poor quality of the received " +
+            "biomaterials(s)."),
```
Comment (mvanniekerkHartwig) [inline patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/qcfail/QCFailReason.java:25]: "biomaterial(s)"

File: patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/qcfail/QCFailReason.java [hunk 3]
```diff
@@ -9,11 +9,29 @@
...
+    HARTWIG_TUMOR_PROCESSING_ISSUE("hartwig_tumor_processing_issue", QCFailType.LOW_QUALITY_BIOPSY, false, QsFormNumber.FOR_082,
+            "Processing failure of tumor analysis", "Whole Genome Sequencing could not be successfully performed on the received " +
+            "tumor biomaterial due to a processing issue."),
+    PIPELINE_FAIL("pipeline_fail", QCFailType.LOW_QUALITY_BIOPSY, false, QsFormNumber.FOR_082,
+            "Insufficient quality of received biomaterial(s)", "Whole Genome Sequencing could not be successfully performed on the " +
+            "received biomaterial(s), this is likely due to poor quality of the received biomaterials(s)."),
```
Comment (mvanniekerkHartwig) [inline patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/qcfail/QCFailReason.java:31]: "biomaterial(s)"

**Review by mvanniekerkHartwig — COMMENTED** [review]
2 small typos

**Pull request merged**